web/src/compose_paste.ts

```diff
@@ -44,8 +44,11 @@ function image_to_zulip_markdown(
     return src ? "[" + title + "](" + src + ")" : (node.getAttribute("alt") ?? "");
 }
 
-// Returns 2 or more if there are multiple valid text nodes in the tree.
-function check_multiple_text_nodes(child_nodes: ChildNode[]): number {
+// Returns the count of valid text nodes in the tree.
+// We return the count as soon as we find at least `cutoff` valid nodes
+// to avoid walking the whole tree.
+// By default, the `cutoff` is a large value.
+function count_valid_text_nodes_upto(child_nodes: ChildNode[], cutoff = 1e9): number {
```

Comment:
I think probably having `cutoff` not have a default would be better, since all existing callers want some limit?